### ISSUE TITLE
[BUG] Unknown host on Milk-V Mars and StarFive VisionFive 2 on Ubuntu 24.10

### ISSUE THREAD

**Issue by FoxKyong** [fastfetch-cli/fastfetch#1391]
*Host identification on Milk-V Mars and StarFive VisionFive 2 on Ubuntu 24.10 is reporting: Unknown Product
* It should identify both hosts since on Ubuntu 24.03 LTS it showed both hosts right
* fastfetch 2.29.0 (riscv), also tried some older versions
* I tried both binaries from github and compiling latest dev from source code

Output of `fastfetch -c ci.jsonc --format json`:
on Milk-V Mars, can't connect to VisionFive2 right now

```
ubuntu@ubuntu-mv:~$ fastfetch -c ci.jsonc
ubuntu@ubuntu-mv                                                                  0.045ms
----------------                                                                  0.134ms
OS: Ubuntu oracular 24.10 riscv64                                                 0.068ms
Host: Unknown Product                                                             0.486ms
Bios (UEFI): 2024.01+dfsg-1ubuntu5 (24.1)                                         0.188ms
Bootmgr: Failed to read efivar: BootCurrent                                       0.062ms
Board: Unknown Product                                                            0.243ms
Chassis: Desktop                                                                  0.225ms
Kernel: Linux 6.11.0-8-generic                                                    0.031ms
Init System: systemd 256.5-2ubuntu3                                               1.218ms
Uptime: 2 hours, 25 mins                                                          0.086ms
Loadavg: 0.07, 0.02, 0.00                                                         0.084ms
Processes: 123                                                                    0.482ms
Packages: 652 (dpkg)                                                              1.563ms
Shell: bash 5.2.32                                                                5.396ms
Editor: $VISUAL or $EDITOR not set                                                0.041ms
Display: Couldn't detect display                                                  7.776ms
Brightness: No result is detected.                                                0.071ms
Monitor: No display detected                                                      0.040ms
LM: sshd 9.7p1 Ubuntu-7ubuntu4 (TTY)                                             11.195ms
DE: No DE found                                                                   0.037ms
WM: No WM found                                                                   0.042ms
WM Theme: WM Theme needs successful WM detection                                  0.034ms
Theme: No themes found                                                            0.320ms
Icons: No icons could be found                                                    0.040ms
Font: No fonts found                                                              0.033ms
Cursor: Couldn't find cursor                                                      0.123ms
Wallpaper: Failed to detect the current wallpaper path                            0.040ms
Terminal: /dev/pts/0                                                              0.275ms
Terminal Font: Unknown terminal: sshd                                             0.036ms
Terminal Size: 90 columns x 27 rows (720px x 418px)                               0.045ms
Terminal Theme: #FCFCFC (FG) - #232627 (BG) [Dark]                               44.370ms
CPU: sifive,u74-mc rv64gc (4) @ 1.50 GHz                                          0.582ms
CPU Cache (L1): 4x32.00 KiB (I), 4x32.00 KiB (D)
CPU Cache (L2): 2.00 MiB (U)                                                      2.356ms
CPU Usage: 0%                                                                   200.363ms
GPU: GPU detection failed                                                         1.530ms
Memory: 337.60 MiB / 7.75 GiB (4%)                                                0.106ms
Physical Memory: Failed to get SMBIOS data                                        0.068ms
Swap: Disabled                                                                    0.078ms
Disk (/): 2.53 GiB / 55.94 GiB (5%) - ext4                                        0.849ms
Btrfs: No btrfs drive found                                                       0.066ms
Zpool: dlopen libzfs.so failed                                                    0.553ms
Battery: No batteries found                                                       0.059ms
Power Adapter: No power adapters found                                            0.057ms
Media Player: No media found                                                      4.660ms
Media: No media found                                                             0.041ms
Public IP: ***************                                         0.138ms
Local IP (end0): 192.168.9.38/24 fe80::6ecf:39ff:fe00:9239/64 (6c:cf:39:00:92:39) [Speed 1 Gbps / MTU 1500] <UP,BROADCAST,RUNNING,MULTICAST,LOWER_UP>                       0.431ms
DNS: 127.0.0.53                                                                   0.088ms
Wifi: No Wifi interfaces found                                                    0.194ms
Date & Time: 2024-11-11 14:05:54                                                  0.087ms
Locale: C.UTF-8                                                                   0.025ms
Vulkan: dlopen libvulkan.so failed                                                0.023ms
OpenGL: dlopen glx failed                                                         0.583ms
OpenCL: dlopen libOpenCL.so failed                                                0.025ms
Users: ubuntu@10.9.3.2 - login time 2024-11-11 13:52:16                           0.193ms
Bluetooth: Failed to call GetManagedObjects                                       2.165ms
Bluetooth Radio: Failed to call GetManagedObjects                                 0.622ms
Sound: Failed to load libpulse.so                                                 0.239ms
Camera: No camera found                                                           0.045ms
Gamepad: No devices detected                                                      0.071ms
Weather: Unknown location; please try *****************                      0.099ms
Network IO (end0): 7.34 KiB/s (IN) - 15.18 KiB/s (OUT)                          708.610ms
Disk IO (mmcblk1): 0 B/s (R) - 0 B/s (W)                                          0.443ms
Physical Disk (mmcblk1): 57.95 GiB [SSD, Fixed]                                   0.532ms
TPM: TPM device is not found                                                      0.050ms
Version: fastfetch 2.29.0 (riscv)                                                 0.023ms
                                                                                  0.006ms
                        
                                                                                  0.045ms
ubuntu@ubuntu-mv:~$ 
```

Output of `fastfetch --list-features`:
```
buntu@ubuntu-mv:~$ fastfetch --list-features
threads
vulkan
wayland
xcb-randr
xcb
xrandr
x11
drm
drm_amdgpu
gio
dconf
dbus
imagemagick6
zlib
xfconf
sqlite3
rpm
egl
glx
osmesa
opencl
libpulse
libddcutil
libelf
libzfs
Directx Headers
linux/videodev2
linux/wireless
```

**Comment by CarterLi** [comment]
Hello. What does `grep -F '' /sys/devices/virtual/dmi/id/*` print?

**Comment by FoxKyong** [comment]
Milk-V Mars:

```
ubuntu@ubuntu-mv:~$ sudo grep -F '' /sys/devices/virtual/dmi/id/*
/sys/devices/virtual/dmi/id/bios_date:01/01/2024
/sys/devices/virtual/dmi/id/bios_release:24.1
/sys/devices/virtual/dmi/id/bios_vendor:U-Boot
/sys/devices/virtual/dmi/id/bios_version:2024.01+dfsg-1ubuntu5
/sys/devices/virtual/dmi/id/board_asset_tag:
/sys/devices/virtual/dmi/id/board_name:Unknown Product
/sys/devices/virtual/dmi/id/board_serial:
/sys/devices/virtual/dmi/id/board_vendor:Unknown
/sys/devices/virtual/dmi/id/board_version:
/sys/devices/virtual/dmi/id/chassis_asset_tag:
/sys/devices/virtual/dmi/id/chassis_serial:
/sys/devices/virtual/dmi/id/chassis_type:3
/sys/devices/virtual/dmi/id/chassis_vendor:Unknown
/sys/devices/virtual/dmi/id/chassis_version:
/sys/devices/virtual/dmi/id/modalias:dmi:bvnU-Boot:bvr2024.01+dfsg-1ubuntu5:bd01/01/2024:br24.1:svnUnknown:pnUnknownProduct:pvr:rvnUnknown:rnUnknownProduct:rvr:cvnUnknown:ct3:cvr:sku:
grep: /sys/devices/virtual/dmi/id/power: Is a directory
/sys/devices/virtual/dmi/id/product_family:
/sys/devices/virtual/dmi/id/product_name:Unknown Product
/sys/devices/virtual/dmi/id/product_serial:MARS-V11-2340-D008E000-00000E71
/sys/devices/virtual/dmi/id/product_sku:
/sys/devices/virtual/dmi/id/product_uuid:5352414d-562d-3131-2d32-3334302d4430
/sys/devices/virtual/dmi/id/product_version:
grep: /sys/devices/virtual/dmi/id/subsystem: Is a directory
/sys/devices/virtual/dmi/id/sys_vendor:Unknown
/sys/devices/virtual/dmi/id/uevent:MODALIAS=dmi:bvnU-Boot:bvr2024.01+dfsg-1ubuntu5:bd01/01/2024:br24.1:svnUnknown:pnUnknownProduct:pvr:rvnUnknown:rnUnknownProduct:rvr:cvnUnknown:ct3:cvr:sku:
```

StarFive VisionFive 2:

```
ubuntu@ubuntu-vf2:~$ sudo grep -F '' /sys/devices/virtual/dmi/id/*
/sys/devices/virtual/dmi/id/bios_date:01/01/2024
/sys/devices/virtual/dmi/id/bios_release:24.1
/sys/devices/virtual/dmi/id/bios_vendor:U-Boot
/sys/devices/virtual/dmi/id/bios_version:2024.01+dfsg-1ubuntu5
/sys/devices/virtual/dmi/id/board_asset_tag:
/sys/devices/virtual/dmi/id/board_name:Unknown Product
/sys/devices/virtual/dmi/id/board_serial:
/sys/devices/virtual/dmi/id/board_vendor:Unknown
/sys/devices/virtual/dmi/id/board_version:
/sys/devices/virtual/dmi/id/chassis_asset_tag:
/sys/devices/virtual/dmi/id/chassis_serial:
/sys/devices/virtual/dmi/id/chassis_type:3
/sys/devices/virtual/dmi/id/chassis_vendor:Unknown
/sys/devices/virtual/dmi/id/chassis_version:
/sys/devices/virtual/dmi/id/modalias:dmi:bvnU-Boot:bvr2024.01+dfsg-1ubuntu5:bd01/01/2024:br24.1:svnUnknown:pnUnknownProduct:pvr:rvnUnknown:rnUnknownProduct:rvr:cvnUnknown:ct3:cvr:sku:
grep: /sys/devices/virtual/dmi/id/power: Is a directory
/sys/devices/virtual/dmi/id/product_family:
/sys/devices/virtual/dmi/id/product_name:Unknown Product
/sys/devices/virtual/dmi/id/product_serial:VF7110B1-2253-D004E000-40000778
/sys/devices/virtual/dmi/id/product_sku:
/sys/devices/virtual/dmi/id/product_uuid:31374656-3031-3142-2d32-3235332d4430
/sys/devices/virtual/dmi/id/product_version:
grep: /sys/devices/virtual/dmi/id/subsystem: Is a directory
/sys/devices/virtual/dmi/id/sys_vendor:Unknown
/sys/devices/virtual/dmi/id/uevent:MODALIAS=dmi:bvnU-Boot:bvr2024.01+dfsg-1ubuntu5:bd01/01/2024:br24.1:svnUnknown:pnUnknownProduct:pvr:rvnUnknown:rnUnknownProduct:rvr:cvnUnknown:ct3:cvr:sku:
```

**Comment by CarterLi** [comment]
It worked on 24.04? What does this command print?

`grep -F '' /sys/firmware/devicetree/base/*`

**Comment by CarterLi** [comment]
> 24.04:
> 
> ```
> ubuntu@ubuntu:~$ grep -F '' /sys/firmware/devicetree/base/*
> grep: /sys/firmware/devicetree/base/#address-cells: binary file matches
> grep: /sys/firmware/devicetree/base/#size-cells: binary file matches
> grep: /sys/firmware/devicetree/base/aliases: Is a directory
> grep: /sys/firmware/devicetree/base/audio-codec: Is a directory
> grep: /sys/firmware/devicetree/base/chosen: Is a directory
> grep: /sys/firmware/devicetree/base/compatible: binary file matches
> grep: /sys/firmware/devicetree/base/cpus: Is a directory
> grep: /sys/firmware/devicetree/base/dvp-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac0-rgmii-rxin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac0-rmii-refin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac1-rgmii-rxin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac1-rmii-refin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gpio-restart: Is a directory
> grep: /sys/firmware/devicetree/base/hdmitx0-pixel-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2srx-bclk-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2srx-lrck-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2stx-bclk-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2stx-lrck-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/mclk-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/memory@40000000: Is a directory
> grep: /sys/firmware/devicetree/base/model: binary file matches
> grep: /sys/firmware/devicetree/base/name: binary file matches
> grep: /sys/firmware/devicetree/base/opp-table-0: Is a directory
> grep: /sys/firmware/devicetree/base/oscillator: Is a directory
> grep: /sys/firmware/devicetree/base/reserved-memory: Is a directory
> grep: /sys/firmware/devicetree/base/rtc-oscillator: Is a directory
> grep: /sys/firmware/devicetree/base/serial-number: binary file matches
> grep: /sys/firmware/devicetree/base/soc: Is a directory
> grep: /sys/firmware/devicetree/base/sound: Is a directory
> grep: /sys/firmware/devicetree/base/stmmac-axi-config: Is a directory
> grep: /sys/firmware/devicetree/base/tdm-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/thermal-zones: Is a directory
> ```
> 
> 24.10:
> 
> ```
> ubuntu@ubuntu-mv:~$ grep -F '' /sys/firmware/devicetree/base/*
> grep: /sys/firmware/devicetree/base/#address-cells: binary file matches
> grep: /sys/firmware/devicetree/base/#size-cells: binary file matches
> grep: /sys/firmware/devicetree/base/aliases: Is a directory
> grep: /sys/firmware/devicetree/base/audio-codec: Is a directory
> grep: /sys/firmware/devicetree/base/chosen: Is a directory
> grep: /sys/firmware/devicetree/base/compatible: binary file matches
> grep: /sys/firmware/devicetree/base/cpus: Is a directory
> grep: /sys/firmware/devicetree/base/dvp-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac0-rgmii-rxin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac0-rmii-refin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac1-rgmii-rxin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gmac1-rmii-refin-clock: Is a directory
> grep: /sys/firmware/devicetree/base/gpio-restart: Is a directory
> grep: /sys/firmware/devicetree/base/hdmitx0-pixel-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2srx-bclk-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2srx-lrck-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2stx-bclk-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/i2stx-lrck-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/mclk-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/memory@40000000: Is a directory
> grep: /sys/firmware/devicetree/base/model: binary file matches
> grep: /sys/firmware/devicetree/base/name: binary file matches
> grep: /sys/firmware/devicetree/base/opp-table-0: Is a directory
> grep: /sys/firmware/devicetree/base/oscillator: Is a directory
> grep: /sys/firmware/devicetree/base/reserved-memory: Is a directory
> grep: /sys/firmware/devicetree/base/rtc-oscillator: Is a directory
> grep: /sys/firmware/devicetree/base/serial-number: binary file matches
> grep: /sys/firmware/devicetree/base/soc: Is a directory
> grep: /sys/firmware/devicetree/base/sound: Is a directory
> grep: /sys/firmware/devicetree/base/stmmac-axi-config: Is a directory
> grep: /sys/firmware/devicetree/base/tdm-ext-clock: Is a directory
> grep: /sys/firmware/devicetree/base/thermal-zones: Is a directory
> ```

Not very useful. What about

`cat /sys/firmware/devicetree/base/*`

**Comment by FoxKyong** [comment]
24.10:

```
ubuntu@ubuntu-mv:~$ cat /sys/firmware/devicetree/base/*
cat: /sys/firmware/devicetree/base/aliases: Is a directory
cat: /sys/firmware/devicetree/base/audio-codec: Is a directory
cat: /sys/firmware/devicetree/base/chosen: Is a directory
milkv,marsstarfive,jh7110cat: /sys/firmware/devicetree/base/cpus: Is a directory
cat: /sys/firmware/devicetree/base/dvp-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac0-rgmii-rxin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac0-rmii-refin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac1-rgmii-rxin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac1-rmii-refin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gpio-restart: Is a directory
cat: /sys/firmware/devicetree/base/hdmitx0-pixel-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2srx-bclk-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2srx-lrck-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2stx-bclk-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2stx-lrck-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/mclk-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/memory@40000000: Is a directory
Milk-V Marscat: /sys/firmware/devicetree/base/opp-table-0: Is a directory
cat: /sys/firmware/devicetree/base/oscillator: Is a directory
cat: /sys/firmware/devicetree/base/reserved-memory: Is a directory
cat: /sys/firmware/devicetree/base/rtc-oscillator: Is a directory
MARS-V11-2340-D008E000-00000E71cat: /sys/firmware/devicetree/base/soc: Is a directory
cat: /sys/firmware/devicetree/base/sound: Is a directory
cat: /sys/firmware/devicetree/base/stmmac-axi-config: Is a directory
cat: /sys/firmware/devicetree/base/tdm-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/thermal-zones: Is a directory
```

24.04:

```
ubuntu@ubuntu:~$ cat /sys/firmware/devicetree/base/*
cat: /sys/firmware/devicetree/base/aliases: Is a directory
cat: /sys/firmware/devicetree/base/audio-codec: Is a directory
cat: /sys/firmware/devicetree/base/chosen: Is a directory
milkv,marsstarfive,jh7110cat: /sys/firmware/devicetree/base/cpus: Is a directory
cat: /sys/firmware/devicetree/base/dvp-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac0-rgmii-rxin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac0-rmii-refin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac1-rgmii-rxin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gmac1-rmii-refin-clock: Is a directory
cat: /sys/firmware/devicetree/base/gpio-restart: Is a directory
cat: /sys/firmware/devicetree/base/hdmitx0-pixel-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2srx-bclk-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2srx-lrck-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2stx-bclk-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/i2stx-lrck-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/mclk-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/memory@40000000: Is a directory
Milk-V Marscat: /sys/firmware/devicetree/base/opp-table-0: Is a directory
cat: /sys/firmware/devicetree/base/oscillator: Is a directory
cat: /sys/firmware/devicetree/base/reserved-memory: Is a directory
cat: /sys/firmware/devicetree/base/rtc-oscillator: Is a directory
MARS-V11-2340-D008E000-00000E71cat: /sys/firmware/devicetree/base/soc: Is a directory
cat: /sys/firmware/devicetree/base/sound: Is a directory
cat: /sys/firmware/devicetree/base/stmmac-axi-config: Is a directory
cat: /sys/firmware/devicetree/base/tdm-ext-clock: Is a directory
cat: /sys/firmware/devicetree/base/thermal-zones: Is a directory
```

**Comment by CarterLi** [comment]
Please test if https://github.com/fastfetch-cli/fastfetch/commit/748b7df0d7c5a81a00f105571cc48b1aca4f98aa fix your issue

**Comment by FoxKyong** [comment]
It works. :)

Milk-V Mars:

```
ubuntu@ubuntu-mv:~/fastfetch/build$ ./fastfetch -c ci.jsonc
ubuntu@ubuntu-mv                                                                        0.063ms
----------------                                                                        0.136ms
OS: Ubuntu oracular 24.10 riscv64                                                       0.106ms
Host: Milk-V Mars                                                                       0.604ms
Bios (UEFI): 2024.01+dfsg-1ubuntu5 (24.1)                                               0.190ms
Bootmgr: Failed to read efivar: BootCurrent                                             0.078ms
Board: mars                                                                             0.153ms
Chassis: Desktop                                                                        0.247ms
Kernel: Linux 6.11.0-8-generic                                                          0.039ms
Init System: systemd 256.5-2ubuntu3                                                    17.432ms
Uptime: 46 mins                                                                         0.110ms
Loadavg: 0.00, 0.00, 0.26                                                               0.086ms
Processes: 122                                                                          0.513ms
Packages: 702 (dpkg)                                                                    0.360ms
Shell: bash 5.2.32                                                                      4.770ms
Editor: $VISUAL or $EDITOR not set                                                      0.039ms
Display: Couldn't detect display                                                        0.133ms
Brightness: No result is detected.                                                      0.065ms
Monitor: No display detected                                                            0.042ms
LM: sshd 9.7p1 Ubuntu-7ubuntu4 (TTY)                                                   11.096ms
DE: No DE found                                                                         0.036ms
WM: No WM found                                                                         0.042ms
WM Theme: WM Theme needs successful WM detection                                        0.033ms
Theme: Theme isn't supported in TTY                                                     0.035ms
Icons: Icons aren't supported in TTY                                                    0.040ms
Font: Font isn't supported in TTY                                                       0.029ms
Cursor: Cursor isn't supported in TTY                                                   0.034ms
Wallpaper: Failed to detect the current wallpaper path                                  0.151ms
Terminal: /dev/pts/0                                                                    0.289ms
Terminal Font: Unknown terminal: sshd                                                   0.038ms
Terminal Size: 96 columns x 28 rows (775px x 423px)                                     0.046ms
Terminal Theme: #FCFCFC (FG) - #232627 (BG) [Dark]                                     15.751ms
CPU: sifive,u74-mc rv64gc (4) @ 1.50 GHz                                                0.577ms
CPU Cache (L1): 4x32.00 KiB (I), 4x32.00 KiB (D)
CPU Cache (L2): 2.00 MiB (U)                                                            2.449ms
CPU Usage: 0%                                                                         200.380ms
GPU: GPU detection failed                                                               0.306ms
Memory: 332.84 MiB / 7.75 GiB (4%)                                                      0.112ms
Physical Memory: Failed to get SMBIOS data                                              0.081ms
Swap: Disabled                                                                          0.078ms
Disk (/): 3.05 GiB / 55.94 GiB (5%) - ext4                                              0.794ms
Btrfs: No btrfs drive found                                                             0.071ms
Zpool: dlopen libzfs.so failed                                                          0.627ms
Battery: No batteries found                                                             0.062ms
Power Adapter: No power adapters found                                                  0.055ms
Media Player: Fastfetch was compiled without DBus support                               0.033ms
Media: Fastfetch was compiled without DBus support                                      0.040ms
Public IP: *************************                                               0.116ms
Local IP (end0): 192.168.9.38/24 fe80::6ecf:39ff:fe00:9239/64 (6c:cf:39:00:92:39) [Speed 1 Gbps / MTU 1500] <UP,BROADCAST,RUNNING,MULTICAST,LOWER_UP>                                   0.448ms
DNS: 127.0.0.53                                                                         0.083ms
Wifi: No Wifi interfaces found                                                          0.192ms
Date & Time: 2024-11-12 19:41:35                                                        0.091ms
Locale: C.UTF-8                                                                         0.036ms
Vulkan: fastfetch was compiled without vulkan support                                   0.036ms
OpenGL: Fastfetch was built without gl support.                                         0.040ms
OpenCL: fastfetch was compiled without OpenCL support                                   0.043ms
Users: ubuntu@192.168.9.10 - login time 2024-11-12 18:58:02                             0.208ms
Bluetooth: Fastfetch was compiled without DBus support                                  0.037ms
Bluetooth Radio: Fastfetch was compiled without DBus support                            0.040ms
Sound: Fastfetch was built without libpulse support                                     0.040ms
Camera: No camera found                                                                 0.043ms
Gamepad: No devices detected                                                            0.067ms
Weather: ****************************                                              0.082ms
Network IO (end0): 7.66 KiB/s (IN) - 17.27 KiB/s (OUT)                                738.644ms
Disk IO (mmcblk1): 0 B/s (R) - 0 B/s (W)                                                0.410ms
Physical Disk (mmcblk1): 57.95 GiB [SSD, Fixed]                                         0.560ms
TPM: TPM device is not found                                                            0.050ms
Version: fastfetch 2.29.0-36 (riscv)                                                    0.020ms
                                                                                        0.006ms
                        
                                                                                        0.044ms
```

StarFive VisionFive:

```
ubuntu@ubuntu-vf2:~/fastfetch/build$ ./fastfetch -c ci.jsonc
ubuntu@ubuntu-vf2                                                                       0.043ms
-----------------                                                                       0.142ms
OS: Ubuntu oracular 24.10 riscv64                                                       0.056ms
Host: StarFive VisionFive 2 v1.3B                                                       0.639ms
Bios (UEFI): 2024.01+dfsg-1ubuntu5 (24.1)                                               0.217ms
Bootmgr: Failed to read efivar: BootCurrent                                             0.076ms
Board: visionfive-2-v1.3b                                                               0.166ms
Chassis: Desktop                                                                        0.298ms
Kernel: Linux 6.11.0-8-generic                                                          0.039ms
Init System: systemd 256.5-2ubuntu3                                                    17.739ms
Uptime: 13 hours, 38 mins                                                               0.101ms
Loadavg: 0.00, 0.03, 0.25                                                               0.079ms
Processes: 123                                                                          0.601ms
Packages: 702 (dpkg)                                                                    0.402ms
Shell: bash 5.2.32                                                                      4.754ms
Editor: $VISUAL or $EDITOR not set                                                      0.041ms
Display: Couldn't detect display                                                        0.129ms
Brightness: No result is detected.                                                      0.065ms
Monitor: No display detected                                                            0.042ms
LM: sshd 9.7p1 Ubuntu-7ubuntu4 (TTY)                                                   10.981ms
DE: No DE found                                                                         0.045ms
WM: No WM found                                                                         0.042ms
WM Theme: WM Theme needs successful WM detection                                        0.032ms
Theme: Theme isn't supported in TTY                                                     0.033ms
Icons: Icons aren't supported in TTY                                                    0.042ms
Font: Font isn't supported in TTY                                                       0.040ms
Cursor: Cursor isn't supported in TTY                                                   0.035ms
Wallpaper: Failed to detect the current wallpaper path                                  0.358ms
Terminal: /dev/pts/0                                                                    0.278ms
Terminal Font: Unknown terminal: sshd                                                   0.036ms
Terminal Size: 96 columns x 28 rows (775px x 423px)                                     0.044ms
Terminal Theme: #FCFCFC (FG) - #232627 (BG) [Dark]                                     13.537ms
CPU: sifive,u74-mc rv64gc (4) @ 1.50 GHz                                                0.607ms
CPU Cache (L1): 4x32.00 KiB (I), 4x32.00 KiB (D)
CPU Cache (L2): 2.00 MiB (U)                                                            3.130ms
CPU Usage: 0%                                                                         200.414ms
GPU: GPU detection failed                                                               0.298ms
Memory: 344.39 MiB / 3.81 GiB (9%)                                                      0.114ms
Physical Memory: Failed to get SMBIOS data                                              0.083ms
Swap: Disabled                                                                          0.083ms
Disk (/): 3.03 GiB / 13.81 GiB (22%) - ext4                                             0.806ms
Btrfs: No btrfs drive found                                                             0.097ms
Zpool: dlopen libzfs.so failed                                                          0.856ms
Battery: No batteries found                                                             0.063ms
Power Adapter: No power adapters found                                                  0.055ms
Media Player: Fastfetch was compiled without DBus support                               0.028ms
Media: Fastfetch was compiled without DBus support                                      0.022ms
Public IP: *****************************                                                0.141ms
Local IP (end0): 192.168.9.42/24 fe80::6ecf:39ff:fe00:6c94/64 (6c:cf:39:00:6c:94) [Speed 1 Gbps / MTU 1500] <UP,BROADCAST,RUNNING,MULTICAST,LOWER_UP>                                   0.479ms
DNS: 127.0.0.53                                                                         0.082ms
Wifi: No Wifi interfaces found                                                          0.259ms
Date & Time: 2024-11-12 19:44:42                                                        0.103ms
Locale: C.UTF-8                                                                         0.023ms
Vulkan: fastfetch was compiled without vulkan support                                   0.036ms
OpenGL: Fastfetch was built without gl support.                                         0.040ms
OpenCL: fastfetch was compiled without OpenCL support                                   0.040ms
Users: ubuntu@192.168.9.10 - login time 2024-11-12 19:02:59                             0.214ms
Bluetooth: Fastfetch was compiled without DBus support                                  0.035ms
Bluetooth Radio: Fastfetch was compiled without DBus support                            0.042ms
Sound: Fastfetch was built without libpulse support                                     0.040ms
Camera: No camera found                                                                 0.045ms
Gamepad: No devices detected                                                            0.064ms
Weather: ****************                                               0.082ms
Network IO (end0): 8.61 KiB/s (IN) - 17.67 KiB/s (OUT)                                738.610ms
Disk IO (mmcblk0): 0 B/s (R) - 0 B/s (W)
Disk IO (mmcblk0boot0): 0 B/s (R) - 0 B/s (W)
Disk IO (mmcblk0boot1): 0 B/s (R) - 0 B/s (W)                                           1.733ms
Physical Disk (mmcblk0): 14.45 GiB [SSD, Fixed]
Physical Disk (mmcblk0boot0): 4.00 MiB [SSD, Fixed, Read-only]
Physical Disk (mmcblk0boot1): 4.00 MiB [SSD, Fixed, Read-only]                          1.315ms
TPM: TPM device is not found                                                            0.057ms
Version: fastfetch 2.29.0-36 (riscv)                                                    0.021ms
                                                                                        0.006ms
                        
                                                                                        0.041ms
```
